### DISCUSSION TITLE
Spurious turning instruction when crossing a specific road

### DISCUSSION THREAD

**Discussion by luziferius** [osmandapp/OsmAnd#11604] [Maps Related]
Hello forum,

On a certain crossroads, I get an invalid routing instruction to "turn left", whereas the rendered route follows the original road.

I’m not sure where the issue lies. The OSM data seems reasonable, so it looks like it’s an issue in OsmAnd’s routing.
One of the segments uses [turn:lanes:both_ways](https://wiki.openstreetmap.org/wiki/Proposed_features/Suffix_both_ways), which may be unsupported by OsmAnd, as that is currently only a proposed suffix.

The related road: https://www.openstreetmap.org/way/628530178.

And here’s the route, as given by OSM directly: https://www.openstreetmap.org/directions?engine=fossgis_osrm_car&route=51.5635%2C7.4087%3B51.5661%2C7.3992#map=17/51.56455/7.40458&layers=N

This involves three segments (in passing order):

1. 
```
lanes=4
lanes:backward=1
lanes:forward=3
turn:lanes:forward=left|through|right
```
2. 
```
lanes=3
lanes:backward=1
lanes:forward=2
turn:lanes:forward=left|through
```
3.
```
lanes=3
lanes:backward=1
lanes:both_ways=1
lanes:forward=1
turn:lanes:backward=through
turn:lanes:both_ways=left
turn:lanes:forward=through
```

When entering a start and end point in OsmAnd similar to the route linked above, it suggests turning left by using the turning lane, while the rendering points forward:

<img alt="Screenshot_20210504-145215_OsmAnd" src="https://user-images.githubusercontent.com/2143820/117009019-98749000-aceb-11eb-913d-ba01bd0b2ada.jpg" width="360">

**Comment by vshcherb** [comment]
That's definitely an issue, sometimes it's hard to say what's wrong in the data, actually everything could be ok but that data is ambiguous, so algorithm doesn't work correctly in all cases. So it's required to have further investigation- https://github.com/osmandapp/OsmAnd/issues/11607

**Reply by luziferius** [reply]
Thanks for the fix by supporting `turn:lanes:both_ways`!

**Reply by vshcherb** [reply]
well.. it's strange that's so widely used and OSM Wiki is not updated. Please fix it if you are interested in it.
